### DISCUSSION TITLE
babel-plugin-react-compiler-runtime with react-native

### DISCUSSION THREAD

**Discussion by kidyfirst** [reactwg/react-compiler#37] [General Questions] [answered]
i have installed the devDependencies

"babel-plugin-react-compiler": "^19.0.0-beta-0dec889-20241115",

and add the plugins in babel config
const ReactCompilerConfig = {}
module.exports = {
  presets: ['module:@react-native/babel-preset'],
  plugins: [
    ['babel-plugin-react-compiler', ReactCompilerConfig],
    'react-native-reanimated/plugin'
 ]
}

but it throw the error
error: Error: Unable to resolve module react/compiler-runtime from ~/apps/mobile/index.js: 

None of these files exist:
  * node_modules/react/compiler-runtime(.android.js|.native.js|.js|.android.jsx|.native.jsx|.jsx|.android.json|.native.json|.json|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx)
  * node_modules/react/compiler-runtime

**Comment by kidyfirst** [comment]
i have added the react-compiler-runtime directly.
const ReactCompilerConfig = {
  target: '18'
};

module.exports = function () {
  return {
    plugins: [
      ['babel-plugin-react-compiler', ReactCompilerConfig],
    ],
  };
};
but issue is still

**Comment by kidyfirst** [comment, marked as answer]
i added the import 'react-compiler-runtime' in entry js file, it is ok now

**Reply by poteto** [reply]
Glad you got it working!
